### PR TITLE
Contacts with persistence and error logging

### PR DESCRIPTION
**Pull request by AlyaNovikova**
- add persistence 
 - make SwiftPM 
 - add error logging

### CODE REVIEW

File: alyanovikova/Contacts/Sources/Contacts/Contacts.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  public init() throws {
+    fileURL = try Self.makeDefaultURL()
+    do {
+      contacts = try [Contact.Id: Contact](jsonFileURL: fileURL)
+    } catch CocoaError.fileReadNoSuchFile {
+      contacts = [:]
+      logger.error("FileURL not found, init with blank ContactBook")
+    } catch let error {
```
Comment (gartemiy) [inline alyanovikova/Contacts/Sources/Contacts/Contacts.swift:38]:
```suggestion
    } catch {
```
Here and in other places

File: alyanovikova/Contacts/Sources/Contacts/Contacts.swift [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    self.name = name
+    self.surname = surname
+    self.phone = phone
+  }
+}
+
+public class ContactBook {
+  public var contacts = [Contact.Id: Contact]()
+  private let fileURL: URL
+  private let logger = Logger(label: "com.google.WearOS.ContactBook")
```
Comment (gartemiy) [inline alyanovikova/Contacts/Sources/Contacts/Contacts.swift:24]: Nit: Maybe replace `WearOS` with something like `Internship` as it is not `WearOS` related.

File: alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      }
+      julia.name = "Julia"
+      ira.name = "Ira"
+      ira.surname = "Hor"
+      ira.phone = "+3801"
+
+      try book.updateContact(
+        newContact: julia)
+      try book.updateContact(
+        newContact: ira)
```
Comment (gartemiy) [inline alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift:106]: not correct alignment I suppose...

File: alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  }
+
+  func testAddUpdateRemoveContact() throws {
+    // WHEN
+    do {
+      let book = try ContactBook()
+      _ = try book.addContact(name: "Alya", surname: "Nov", phone: "+7911")
+      _ = try book.addContact(name: "Juliaaaaa", surname: "Sam", phone: "+3800")
+      _ = try book.addContact(name: "Nil", surname: "Nil", phone: "Nil")
+    }
```
Comment (gartemiy) [inline alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift:68]: I think it is a GIVEN block

File: alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      XCTAssertEqual(book.listContacts(where: { $0.name == "Alya" }).count, 0)
+      XCTAssertEqual(
+        book.listContacts(where: {
+          guard $0.phone.count > 0 else {
+            return false
+          }
+          return $0.phone[$0.phone.startIndex] == "+"
+        }).count, 2)
+    }
+  }
```
Comment (gartemiy) [inline alyanovikova/Contacts/Tests/ContactsTests/ContactsTests.swift:160]: Why not 4 different tests?

File: alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import XCTest
+
+#if !canImport(ObjectiveC)
+public func allTests() -> [XCTestCaseEntry] {
+    return [
+        testCase(ContactsTests.allTests),
+    ]
+}
+#endif
```
Comment (gartemiy) [inline alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift:9]: needs swift-format?

File: alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import XCTest
+
+#if !canImport(ObjectiveC)
+public func allTests() -> [XCTestCaseEntry] {
+    return [
+        testCase(ContactsTests.allTests),
```
Comment (gartemiy) [inline alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift:6]: You have already removed `allTests`, haven't you?

File: alyanovikova/Contacts/Package.swift [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  name: "Contacts",
+  products: [
+    // Products define the executables and libraries produced by a package, and make them visible to other packages.
+    .library(
+      name: "Contacts",
+      targets: ["Contacts"]),
+  ],
+  dependencies: [
+    // Dependencies declare other packages that this package depends on.
+    // .package(url: /* package url */, from: "1.0.0"),
```
Comment (gartemiy) [inline alyanovikova/Contacts/Package.swift:16]: You may remove these comments as it is clear you already know how to use them.

File: alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift (hunk 7 above)
Comment (AlyaNovikova) [reply alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift:6]: I didn't really understand what needs to be fixed here 🤔

Comment (gartemiy) [reply alyanovikova/Contacts/Tests/ContactsTests/XCTestManifests.swift]: Indentation ;)

**Comment by AlyaNovikova** [comment]
fixed!

**Pull request merged**